pymeasure/instruments/agilent/agilentN9320A.py

```diff
@@ -0,0 +1,296 @@
+        average mode should be on."""
+        self.write("SENS:AVER:COUN %d" % strict_range(value, [1, 1000]))
+
+    def average_on(self):
+        self.write("SENS:AVER:STAT ON")
+        """Activate the averaging."""
+
+    def average_off(self):
+        """De-activate the averaging."""
+        self.write("SENS:AVER:STAT OFF")
+
+    def set_peak_max(self):
+        """Peak search method: the max value of the trace is taken. Next peak
+        commands always use threshold and excursion method."""
+        self.write("CALC:MARK:PEAK:SEAR:MODE MAX")
+
+    def set_peak_par(self, number=1, threshold=PEAK_TH, excursion=PEAK_EXC):
+        """Peak search method using threshold and excursion."""
+        self.write("CALC:MARK:PEAK:SEAR:MODE PAR")
+        self.write("CALC:MARK:PEAK:THR:EXC %.2f" % excursion)
+        self.write("CALC:MARK%s:PEAK:THR %.2f" % (number, threshold))
+        self.write("CALC:MARK:PEAK:THR:STATE 1")
+
+    def peak_search(self, number=1):
+        """Peak search: the search method can be set with set_peak_par or
+        set_peak_max."""
+        self.write("CALC:MARK%s:MAX" % number)
+
+    def next_peak_right(self,number=1):
+        """ Search the next peak on the right of the actual position of the
+        marker. It always use threshold and excursion. """
+        self.write("CALC:MARK%s:MAX:RIGHT" %number)
+
+    def next_peak_left(self,number=1):
+        """ Search the next peak on the left of the actual position of the
+        marker. It always use threshold and excursion. """
+        self.write("CALC:MARK%s:MAX:LEFT" %number)
+
+    def peak_exist(self,number=1):
+        """ Check if there is a peak in the trace. It has to be used after
+        peak_search, next_peak_left or next_peak_right. If no error it returns
+        [freq, amplitude] of the cursor number-th. If error it returns
+        [NAN, NAN]. """
+        if self.error_check('780 No Peak Found'):
+            log.info('No peak found!')
+            return False, [[float('NAN'), float('NAN')]]
+        else:
+            x = float(self.marker_x(number))
+            y = float(self.marker_y(number))
+            log.info('Peak found at %g Hz, amplitude %.4e %s' % (x, y, self.unit))
+            return True, [[x ,y]]
+
+    def peak_output(self, number=1, avg=5, center=False, lr=False):
+        """ Returns the frequency and the intensity of the highest peak.
+        Search method can be set with  set_peak_par or set_peak_max,
+        before calling this method.
+        It can center the central frequency to the central peak.
+        It can also look the first peaks at both left and right
+        around the highest one. """
+
+        self.init_single()
+        self.average_on()
+        self.average_number(avg)
+
+        sleep(self.DELAY)
+
+        if self.adapter.connection.timeout < (avg*self.D_FACTOR*self.sweep_time)*1000:
+            self.adapter.connection.timeout = (avg*self.D_FACTOR*self.sweep_time)*1000
```

Comment:
Is this timeout modification purely for the opc call? If yes, it should be reset to the original value after you are finished, otherwise you permanently set the timeout.